spec/webvalve/service_url_converter_spec.rb

```diff
@@ -0,0 +1,149 @@
+require 'spec_helper'
+
+RSpec.describe WebValve::ServiceUrlConverter do
+  let(:url) { "http://bar.com" }
+
+  subject { described_class.new(url: url) }
+
+  describe '#regexp' do
+    it "returns a regexp" do
+      expect(subject.regexp).to be_a(Regexp)
+    end
+
+    context "with a trailing *" do
+      let(:url) { "http://bar.com/*" }
+
+      it "matches when empty" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches when existing" do
+        expect("http://bar.com/foobaloo").to match(subject.regexp)
+      end
+
+      it "matches with additional tokens" do
+        expect("http://bar.com/foobaloo/wink").to match(subject.regexp)
+      end
+
+      it "doesn't match when missing the trailing slash tho" do
+        expect("http://bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a totally wildcarded protocol" do
+      let(:url) { "*://bar.com" }
+
+      it "matches http" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches anything else" do
+        expect("gopher://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches empty" do
+        expect("://bar.com").to match(subject.regexp)
```

Comment:
Here's another example of an invalid URL that'd match, but I think that's fine. Codifying because I wanted to make sure that removing `\b` from the word boundary chars didn't change behavior at the edge of the string.

Reply:
yea i raised a question about this above. i think i'm fine with either option but i'm curious if you have a strong preference for supporting the 0 char match

Reply:
I do have that pref, alluded to in other thread :)